run/sigterm-handler/main.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Sample run-helloworld is a minimal Cloud Run service.
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+// [START cloudrun_sigterm_handler]
+func main() {
+	// Determine port for HTTP service.
+	port := os.Getenv("PORT")
+	if port == "" {
+		port = "8080"
+		log.Printf("defaulting to port %s", port)
+	}
+
+	srv := &http.Server{
+		Addr:    ":" + port,
+		Handler: http.HandlerFunc(handler),
+	}
+
+	// Create channel to listen for signals
+	signalChan := make(chan os.Signal, 1)
+	// SIGINT handles Ctrl+C locally
+	// SIGTERM handles Cloud Run termination signal
+	signal.Notify(signalChan, syscall.SIGINT, syscall.SIGTERM)
+
+	// Start HTTP server.
+	go func() {
+		log.Printf("listening on port %s", port)
+		if err := srv.ListenAndServe(); err != nil {
+			log.Fatal(err)
```

Comment:
I don't recommend blindly Fataling here.
Main routine will shut down this server quickly.
Then, without completing the rest of the log stmts in main routine, it might come here and Fatal (indicating error).

Err here can actually be http.ErrServerClosed, and in that case the program should not be terminated right away, because we cannot clean db connections etc. as you described at the end of the program.

Reply:
done.